snowflake/cmd/policyRecommendation.go

```diff
@@ -0,0 +1,285 @@
+	if startTime != "" {
+		query += fmt.Sprintf(`AND
+  flowStartSeconds >= '%s'
+`, startTime)
+	}
+
+	var endTime string
+	if endTs != "" {
+		endTime = endTs
+	} else if end != "" {
+		var err error
+		endTime, err = timestamps.ParseTimestamp(end, now)
+		if err != nil {
+			return "", err
+		}
+	}
+	if endTime != "" {
+		query += fmt.Sprintf(`AND
+  flowEndSeconds >= '%s'
+`, endTime)
+	}
+
+	if clusterUUID != "" {
+		_, err := uuid.Parse(clusterUUID)
+		if err != nil {
+			return "", err
+		}
+		query += fmt.Sprintf(`AND
+  clusterUUID = '%s'
+`, clusterUUID)
+	} else {
+		logger.Info("No clusterUUID input, all flows will be considered during policy recommendation.")
+	}
+
+	query += `GROUP BY
+sourcePodNamespace,
+sourcePodLabels,
+destinationIP,
+destinationPodNamespace,
+destinationPodLabels,
+destinationServicePortName,
+destinationTransportPort,
+protocolIdentifier,
+flowType
+	`
+
+	if limit > 0 {
+		query += fmt.Sprintf(`
+LIMIT %d`, limit)
+	} else {
+		// limit the number unique flow records to 500k to avoid udf timeout
+		query += `
+LIMIT 500000`
+	}
+
+	// Choose the destinationIP as the partition field for the preprocessing
+	// UDTF because flow rows could be divided into the most subsets
+	functionName = udfs.GetFunctionName(preprocessingFunctionName, functionVersion)
+	query += fmt.Sprintf(`), processed_flows AS (SELECT r.appliedTo, r.ingress, r.egress FROM filtered_flows AS f,
+TABLE(%s(
+	'%s',
+	%d,
+	'%s',
+	'%s',
+  f.sourcePodNamespace,
+  f.sourcePodLabels,
+  f.destinationIP,
+  f.destinationPodNamespace,
+  f.destinationPodLabels,
+  f.destinationServicePortName,
+  f.destinationTransportPort,
+  f.protocolIdentifier,
+  f.flowType
+) over (partition by f.destinationIP)) as r
+`, functionName, jobType, isolationMethod, nsAllowList, labelIgnoreList)
+
+	// Scan the row number for each appliedTo group and divide the partitions
+	// larger than partitionSizeLimit.
+	query += fmt.Sprintf(`), pf_with_index AS (
+SELECT 
+  pf.appliedTo, 
+  pf.ingress, 
+  pf.egress, 
+  floor((Row_number() over (partition by pf.appliedTo order by egress))/%d) as row_index 
+FROM processed_flows as pf
+`, partitionSizeLimit)
+
+	// Choose the appliedTo as the partition field for the policyRecommendation
+	// UDTF because each network policy is recommended based on all ingress and
+	// egress traffic related to an appliedTo group.
+	functionName = udfs.GetFunctionName(policyRecommendationFunctionName, functionVersion)
+	query += fmt.Sprintf(`) SELECT r.jobType, r.recommendationId, r.timeCreated, r.yamls FROM pf_with_index,
+TABLE(%s(
+  '%s',
+  '%s',
+  %d,
+  '%s',
+  pf_with_index.appliedTo,
+  pf_with_index.ingress,
+  pf_with_index.egress
+) over (partition by pf_with_index.appliedTo, pf_with_index.row_index)) as r
+`, functionName, jobType, recommendationID, isolationMethod, nsAllowList)
+
+	return query, nil
+}
+
+// policyRecommendationCmd represents the policy-recommendation command
+var policyRecommendationCmd = &cobra.Command{
```

Comment:
do you think the command should be "recommend-policies" or "run policy-recommendation" for consistency with other commands? I don't remember what the command is for the ClickHouse backend, is it the same?

Reply:
I was trying to keep consistent with the ClickHouse side (`theia policy-recommendation run/status/retrieve/..`), although we don't have any sub-commands for now.

Reply:
sounds good, let's keep it consistent with the other implementation then